app/models/concerns/pxe_loader_support.rb

```diff
@@ -28,6 +29,7 @@ def all_loaders_map(precision = 'x64', httpboot_host = "httpboot_host")
         "Grub2 ELF" => "grub2/grub#{precision}.elf",
         "Grub2 UEFI" => "grub2/grub#{precision}.efi",
         "Grub2 UEFI SecureBoot" => "grub2/shim#{precision}.efi",
+        "Grub2 UEFI SecureBoot (target OS)" => "grub2/@@subdir@@/shim#{precision}.efi",
```

Comment:
I wonder if we can somehow guard this on the capability I spoke about in https://github.com/theforeman/smart-proxy/pull/877#pullrequestreview-1689754274 so the user can only select it if it's supported.

Reply:
Thanks @ekohl, this would be great! I will read into that topic and have a look how I can guard this.

Reply:
Hey @ekohl, I had a deeper look at the changes that would be necessary to add the new PXE loader to the list of PXE loaders depending on the smart proxy capability. According to my findings, this would require changes in more than 15 additional files.
In the Web UI the list of PXE loaders is loaded when an operating system is selected in the Host Edit page. As the Smart Proxy depends on the Subnet, we would have to pass the currently selected Subnet down from that point through [operatingsystem.rb](https://github.com/theforeman/foreman/blob/develop/app/models/operatingsystem.rb) (and all derived classes) to the point where the dictionary of PXE loaders is assembled in [pxe_loader_support.rb](https://github.com/theforeman/foreman/blob/develop/app/models/concerns/pxe_loader_support.rb). 
Furthermore we would also have to guard the available PXE loaders in the parameter validation in the [hostgroups_controller.rb](https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/hostgroups_controller.rb) and [hosts_controller.rb](https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/hosts_controller.rb) (both for parameter validations). And there are still some more references to check for required changes.
We would also have to reload the PXE loaders in case the Subnet (and therefore the Smart Proxy) is changed and handle cases where the new PXE loader isn’t available any more after the change of the Subnet. I have not identified yet what would be necessary for that. But this also means, that we cannot freeze the dictionary of PXE loaders for each operating system upon creation anymore, as we have to regenerate it on every Subnet change.

All in all this seems a lot of effort for guarding the list of available PXE loaders depending on the Smart Proxy capability. So I looked for an alternative solution:

I now was thinking about guarding the new PXE loader at the moment the changes are submitted in the Web UI and in the actual handling of the API requests. What do you think?

Reply:
The behavior has completely changed with our new approach. The Smart Proxies capabilities will be checked in module `Orchestration::DHCP` when the boot file name is set in the DHCP attributes. If a Smart Proxy does not provide the required capability the legacy boot file names will be used.